pkg/kv/kvserver/logstore/logstore.go

```diff
@@ -360,8 +360,13 @@ var nonBlockingSyncWaiterCallbackPool = sync.Pool{
 	New: func() interface{} { return new(nonBlockingSyncWaiterCallback) },
 }
 
-var valPool = sync.Pool{
-	New: func() interface{} { return &roachpb.Value{} },
+var logAppendPool = sync.Pool{
+	New: func() interface{} {
+		return new(struct {
+			roachpb.Value
+			enginepb.MVCCStats
+		})
```

Comment:
nit: may as well give this type a name, but don't really care either way.

Reply:
Meh, I think that's one of the cases where it's somewhat beneficial to not introduce a type explicitly. There is exactly one caller/user of this type, and it's tied to the pool, and the pool is tied to a single function (also a reason why I renamed the pool after the function's name).